### PR TITLE
Refactoring: prefixing object store related classes with Os

### PR DESCRIPTION
**Pull request by kneth**
Prefixing/renaming of:

* `io.realm.internal.SharedRealm` -> `io.realm.internal.OsSharedRealm`
* `io.realm.internal.Collection` -> `io.realm.internal.OsResults`
* `io.realm.internal.CollectionChangeSet` -> `io.realm.internal.OsCollectionChangeSet`

See also: #4215

### CODE REVIEW

File: realm/realm-library/src/androidTest/java/io/realm/UnManagedOrderedRealmOsResultsTests.java [hunk 1]
```diff
@@ -53,7 +53,7 @@
  * # List
  *
  *  - void add(int location, E object);
- *  - boolean addAll(int location, Collection<? extends E> collection);
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/UnManagedOrderedRealmOsResultsTests.java]: Renames in this class are also wrong

File: realm/realm-library/src/androidTest/java/io/realm/UnManagedRealmOsResultsTests.java [hunk 2]
```diff
@@ -38,10 +38,10 @@
  * Test class for all methods part of the the {@link RealmCollection} interface, that have a different behavior
  * than managed RealmCollection classes.
  *
- * See {@link ManagedRealmCollectionTests} for similar tests for the managed behavior.
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/UnManagedRealmOsResultsTests.java]: And this in this class

File: realm/realm-library/src/androidTest/java/io/realm/internal/OsResultsTests.java [hunk 3]
```diff
@@ -0,0 +1,534 @@
+/*
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/internal/OsResultsTests.java]: It seems that renaming this killed the git history so the old file is deleted and this is created. That doesn't seem ideal?

File: realm/realm-library/src/androidTest/java/io/realm/internal/SharedRealmTests.java [hunk 4]
```diff
@@ -43,165 +43,165 @@
     public final ExpectedException thrown = ExpectedException.none();
 
     private RealmConfiguration config;
-    private SharedRealm sharedRealm;
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/internal/SharedRealmTests.java]: Shouldn't this file be renamd to `OsSharedRealmTests` ?

File: realm/realm-library/src/main/java/io/realm/OrderedCollectionChangeSet.java [hunk 5]
```diff
@@ -17,31 +17,31 @@
 package io.realm;
 
 /**
- * This interface describes the changes made to a collection during the last update.
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/OrderedCollectionChangeSet.java]: Rename is wrong in this class

File: realm/realm-library/src/main/java/io/realm/OrderedRealmCollection.java [hunk 6]
```diff
@@ -20,7 +20,7 @@
 
 
 /**
- * An {@code OrderedRealmCollection} is a collection which maintains an ordering for its elements. Every
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/OrderedRealmCollection.java]: Same, renames are wrong in this class.

File: realm/realm-library/src/main/java/io/realm/OrderedRealmCollectionChangeListener.java [hunk 7]
```diff
@@ -18,7 +18,7 @@
 
 /**
  * {@link OrderedRealmCollectionChangeListener} can be registered with a {@link RealmResults} to receive a notification
- * with a {@link OrderedCollectionChangeSet} to describe the details of what have been changed in the collection from
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/OrderedRealmCollectionChangeListener.java]: Same, class should be reverted.

File: realm/realm-library/src/main/java/io/realm/OrderedRealmCollectionSnapshot.java [hunk 8]
```diff
@@ -199,12 +199,12 @@ public boolean deleteFirstFromRealm() {
...
         realm.checkIfValidAndInTransaction();
-        UncheckedRow row = collection.lastUncheckedRow();
-        return row != null && row.isAttached() && collection.deleteLast();
+        UncheckedRow row = osResults.lastUncheckedRow();
+        return row != null && row.isAttached() && osResults.deleteLast();
     }
 
     /**
-     * This deletes all objects in the collection from the underlying Realm. All objects in the collection snapshot
+     * This deletes all objects in the osResults from the underlying Realm. All objects in the osResults snapshot
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/OrderedRealmCollectionSnapshot.java]: Wrong rename.

File: realm/realm-library/src/main/java/io/realm/Realm.java [hunk 9]
```diff
@@ -1094,7 +1094,7 @@ private Scanner getFullStringScanner(InputStream in) {
     }
 
     /**
-     * Copies a collection of RealmObjects to the Realm instance and returns their copy. Any further changes to the
+     * Copies a osResults of RealmObjects to the Realm instance and returns their copy. Any further changes to the
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/Realm.java]: Wrong rename

File: realm/realm-library/src/main/java/io/realm/Realm.java [hunk 10]
```diff
@@ -1104,7 +1104,7 @@ private Scanner getFullStringScanner(InputStream in) {
      * @param objects the RealmObjects to copy to the Realm.
      * @return a list of the the converted RealmObjects that all has their properties managed by the Realm.
      * @throws io.realm.exceptions.RealmException if any of the objects has already been added to Realm.
-     * @throws java.lang.IllegalArgumentException if any of the elements in the input collection is {@code null}.
+     * @throws java.lang.IllegalArgumentException if any of the elements in the input osResults is {@code null}.
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/Realm.java]: Same

File: realm/realm-library/src/main/java/io/realm/RealmCollection.java [hunk 11]
```diff
@@ -24,19 +24,19 @@
 
 
 /**
- * {@code RealmCollection} is the root of the collection hierarchy that Realm supports. It defines operations on data
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/RealmCollection.java]: Renames are all wrong in this class.

File: realm/realm-library/src/main/java/io/realm/RealmList.java [hunk 12]
```diff
@@ -355,9 +356,9 @@ public E remove(int location) {
     }
 
     /**
-     * Removes one instance of the specified object from this {@code Collection} if one
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/RealmList.java]: All comments in this class are wrong.

File: realm/realm-library/src/main/java/io/realm/RealmResults.java [hunk 13]
```diff
@@ -124,12 +123,12 @@ public boolean isLoaded() {
      */
     @Override
     public boolean load() {
-        // The Collection doesn't have to be loaded before accessing it if the query has not returned.
-        // Instead, accessing the Collection will just trigger the execution of query if needed. We add this flag is
+        // The OsResults doesn't have to be loaded before accessing it if the query has not returned.
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/RealmResults.java]: Comments should still use `Collection`

File: realm/realm-library/src/objectServer/java/io/realm/UserStore.java [hunk 14]
```diff
@@ -65,7 +65,7 @@
     /**
      * Returns a collection of all users saved in the User store.
      *
-     * @return Collection of all users. If no users exist, an empty collection is returned.
```
Comment (cmelchior) [inline realm/realm-library/src/objectServer/java/io/realm/UserStore.java]: Wrong rename

**Comment by kneth** [comment]
@cmelchior Yeah, `Collection` is a dangerous word 😢

**Comment by kneth** [comment]
Closing, and will do it in another PR

**Pull request closed without merging**